src/zenml/integrations/modal/step_operators/modal_step_operator.py

```diff
@@ -0,0 +1,217 @@
+        """Get the Docker build configurations for the Modal step operator.
+
+        Args:
+            deployment: The pipeline deployment.
+
+        Returns:
+            A list of Docker build configurations.
+        """
+        builds = []
+        for step_name, step in deployment.step_configurations.items():
+            if step.config.step_operator == self.name:
+                build = BuildConfiguration(
+                    key=MODAL_STEP_OPERATOR_DOCKER_IMAGE_KEY,
+                    settings=step.config.docker_settings,
+                    step_name=step_name,
+                )
+                builds.append(build)
+
+        return builds
+
+    def launch(
+        self,
+        info: "StepRunInfo",
+        entrypoint_command: List[str],
+        environment: Dict[str, str],
+    ) -> None:
+        """Launch a step run on Modal.
+
+        Args:
+            info: The step run information.
+            entrypoint_command: The entrypoint command for the step.
+            environment: The environment variables for the step.
+
+        Raises:
+            subprocess.CalledProcessError: If the modal app exits with a non-zero code.
+            ValueError: If no Docker credentials are found for the container registry.
+        """
+        settings = cast(ModalStepOperatorSettings, self.get_settings(info))
+        image_name = info.get_image(key=MODAL_STEP_OPERATOR_DOCKER_IMAGE_KEY)
+        zc = Client()
+        stack = zc.active_stack
+
+        if not stack.container_registry:
+            raise ValueError(
+                "No Container registry found in the stack. "
+                "Please add a container registry and ensure "
+                "it is correctly configured."
+            )
+
+        if docker_creds := stack.container_registry.credentials:
+            docker_username, docker_password = docker_creds
+        else:
+            raise RuntimeError(
+                "No Docker credentials found for the container registry."
+            )
+
+        my_secret = modal.secret._Secret.from_dict(
+            {
+                "REGISTRY_USERNAME": docker_username,
+                "REGISTRY_PASSWORD": docker_password,
+            }
+        )
+
+        spec = modal.image.DockerfileSpec(
+            commands=[f"FROM {image_name}"], context_files={}
+        )
+
+        zenml_image = (
+            modal.Image._from_args(
+                dockerfile_function=lambda *_, **__: spec,
+                force_build=False,
+                image_registry_config=modal.image._ImageRegistryConfig(
+                    api_pb2.REGISTRY_AUTH_TYPE_STATIC_CREDS, my_secret
+                ),
+            )
+            .env(environment)
+            .run_commands(" ".join(entrypoint_command))
+        )
+
+        resource_settings = info.config.resource_settings
+        gpu_str = settings.gpu
+        if resource_settings.gpu_count is not None:
+            gpu_str += f":{resource_settings.gpu_count}"
```

Comment:
`gpu_str` is coming from the settings and might be optional right? In that case, the `+=` will fail I think?

Reply:
But I have a conditional around it to catch it? Shouldn't that be enough?

Reply:
Yes now it's fine. When I commented there was a conditional around the gpu_count but not the actual gpu_str.

Now one last question: If someone just specifies the `gpu_count`, this string will be `:2` for example. Is that a valid value for modal, or should it only be appended if the type was specified?

```python
gpu_str = settings.gpu
if gpu_str and resource_settings.gpu_count is not None:
  gpu_str += f":{resource_settings.gpu_count}"
```

Reply:
Yes good point. I should check for that. Thanks!

Reply:
Done